### PR TITLE
[add] affinity field to istiod

### PR DESCRIPTION
**Pull request by ymktmk**
**Please provide a description of this PR:**

This is an addition to helm Chart's value.yaml !
Even though the affinity field is actually provided, there was no field of it in values.yaml, so I added it.

https://github.com/istio/istio/blob/master/manifests/charts/istio-control/istio-discovery/templates/deployment.yaml#L65~L68

### CODE REVIEW

**Review by morvencao — COMMENTED** [review]
/ok-to-test

**Comment by ymktmk** [comment]
@morvencao 
Please add a `release-notes-none` label .

**Comment by jacob-delgado** [comment]
Since this existed in the chart this LGTM, but please add a release note.

File: releasenotes/notes/add-affinity-field-to-istiod.yaml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+apiVersion: release-notes/v2
+kind: feature
+area: traffic-management
```
Comment (zirain) [inline releasenotes/notes/add-affinity-field-to-istiod.yaml:3]:
```suggestion
area: installation
```

Comment (ymktmk) [reply releasenotes/notes/add-affinity-field-to-istiod.yaml]: I accept that change!

**Review by zirain — APPROVED** [review]

File: manifests/charts/istio-control/istio-discovery/values.yaml [hunk 2]
```diff
@@ -32,6 +32,8 @@ pilot:
 
   env: {}
 
+  affinity: {}
```
Comment (hanxiaop) [inline manifests/charts/istio-control/istio-discovery/values.yaml:35]: I think tolerations were added alongside affinity, and neither was updated in the IstioOperator file. Should we add that as well?

https://github.com/istio/istio/blob/9852844f311e27b55ffaa0ac71b8968b44b453eb/manifests/charts/istio-control/istio-discovery/templates/deployment.yaml#L69-L71

Comment (ymktmk) [reply manifests/charts/istio-control/istio-discovery/values.yaml:35]: Should I add `tolerations: []` in [manifests/charts/istio-control/istio-discovery/values.yaml](https://github.com/istio/istio/pull/47639/files/a3399f3d2159f4baae63dd342db267d8becc00c8#diff-72aceeff43207b39e2ff9d8d20cb5e26eafe3c7a07d3e648099887b3c47f549e) File ?

Comment (jacob-delgado) [reply manifests/charts/istio-control/istio-discovery/values.yaml:35]: I'd open another PR for that.

**Review by jacob-delgado — APPROVED** [review]

**Pull request merged**